src/editor/components/LessonSection.js

```diff
@@ -1,32 +1,177 @@
-import React from 'react';
+import React, { useState } from 'react';
 import LessonPlanTextInput from './LessonPlanTextInput';
-import { Text, View, SafeAreaView, StyleSheet } from 'react-native';
+import { Text, View, SafeAreaView, Platform, StyleSheet, TextInput } from 'react-native';
+import { useDispatch } from 'react-redux';
+import {
+  addToSection,
+  removeFromSection,
+} from '../../services/editor/lessonPlanSlice';
+import store from '../../services/configureStore';
+
+const LessonSection = ({ sectionType, subtitle }) => {
+  const [sectionContent, setSectionContent] = useState([]);
+  // const sectionContent = [
+  //   {
+  //     section: "warmup",
+  //     content: "warmup content",
+  //   }
+  // ]
+  const [sectionActivityCards, setSectionActivityCards] = useState([]);
+  const [clicked, setClicked] = useState(false);
+  const dispatch = useDispatch();
+  const handleClick = () => {
+    setClicked(true);
+  };
+
+  const ContentCard = () => {
+    return (
+      clicked && (
+        <View style={styles.ContentCardStyle}>
+          <TextInput
+            multiline
+            placeholder={'Add Text'}
+            returnKeyType="next"
+            onSubmitEditing={e => {
+              if (e.nativeEvent.text) {
+                setSectionContent([...sectionContent, e.nativeEvent.text]);
+                store.dispatch(
+                  addToSection({
+                    type: 'text',
+                    section: sectionType,
+                    content: e.nativeEvent.text,
+                  })
+                );
+              }
+              setClicked(false);
+            }}
+          />
+        </View>
+      )
+    );
+  };
+
+  const StoredContent = ({ text, index }) => {
+    const [newText, setnewText] = useState(text);
+    return (
+      <View style={styles.ContentCardStyle}>
+        <TextInput
+          multiline
+          defaultValue={text}
+          onChangeText={txt => setnewText(txt)}
+          returnKeyType="next"
+          onSubmitEditing={e => {
+            //if (!e.nativeEvent.text) {
+            setSectionContent(state => {
+              newContent = [...state];
+              //delete the content from section content state when empty
+              newContent = newContent.filter((_, i) => i != index);
+              store.dispatch(
+                removeFromSection({
+                  type: 'text',
+                  section: sectionType,
+                  content: text,
+                })
+              );
+              // if the content isn't empty, add the edited content to store and newContent
+              if (e.nativeEvent.text) {
+                newContent[index] = e.nativeEvent.text;
+                store.dispatch(
+                  addToSection({
+                    type: 'text',
+                    section: sectionType,
+                    content: e.nativeEvent.text,
+                  })
+                );
+              }
```

Comment:
Ok I'm quite confused what's going on here. Isn't this a duplicate of what's going on in `ContentCard` [here](https://github.com/uoftblueprint/sistema/blob/7adecbf92eef16a7b421f7eca30117d1397850e3/src/editor/components/LessonSection.js#L36)? Why are we dispatching content to the redux store a second time?